File: 23 - Validate Palindrome Permutation/solution.js
function canFormPalindrome(str) {
  const freq = {};
  const cleaned = str.toLowerCase().replace(/[^a-z]/g, "");

  for (let char of cleaned) {
    freq[char] = (freq[char] || 0) + 1;
  }

  let oddCount = 0;

  for (let count of Object.values(freq)) {
    if (count % 2 !== 0) oddCount++;
    if (oddCount > 1) return false;
  }

  return true;
}

//   Test cases
console.log(canFormPalindrome("Tact Coa"));

File: 23 - Validate Palindrome Permutation/solution.py
def can_form_palindrome(str):
    # Your code here

    cleaned = str.lower().replace(" ", "").replace("\n", "").replace("\t", "")
    freq = {}

    for char in cleaned:
        freq[char] = freq.get(char, 0) + 1

    odd_count = 0

    for count in freq.values():
        if count % 2 != 0:
            odd_count += 1

        if odd_count > 1:
            return False
    return True


#   Test cases
print(can_form_palindrome("Tact Coa"))



